core/trino-main/src/test/java/io/trino/server/security/oauth2/TestOAuth2WebUiAuthenticationFilter.java

```diff
@@ -117,12 +117,13 @@ public void setup()
                         .put("http-server.https.enabled", "true")
                         .put("http-server.https.keystore.path", Resources.getResource("cert/localhost.pem").getPath())
                         .put("http-server.https.keystore.key", "")
+                        .put("http-server.authentication.oauth2.issuer", "https://localhost:4444/")
                         .put("http-server.authentication.oauth2.auth-url", idpUrl + "/oauth2/auth")
                         .put("http-server.authentication.oauth2.token-url", idpUrl + "/oauth2/token")
                         .put("http-server.authentication.oauth2.jwks-url", idpUrl + "/.well-known/jwks.json")
                         .put("http-server.authentication.oauth2.client-id", TRINO_CLIENT_ID)
                         .put("http-server.authentication.oauth2.client-secret", TRINO_CLIENT_SECRET)
-                        .put("http-server.authentication.oauth2.audience", TRINO_AUDIENCE)
+                        .put("http-server.authentication.oauth2.additional-audiences", TRUSTED_CLIENT_ID)
```

Comment:
In this case Trino's client will accept not only ID tokens issued for it but also for `TRUSTED_CLIENT_ID` which is against the spec:
> The Client MUST validate that the aud (audience) Claim contains its client_id value registered at the Issuer identified by the iss (issuer) Claim as an audience. The aud (audience) Claim MAY contain an array with more than one element. The ID Token MUST be rejected if the ID Token does not list the Client as a valid audience, or if it contains additional audiences not trusted by the Client.

Reply:
Already address in your other comment for the same issue